jenkins/automation/rgw/ceph/ceph-deploy.groovy

```diff
@@ -0,0 +1,153 @@
+pipeline {
+    agent {
+        node {
+            label 'docker-k8-deployment-node'
+        }
+    }
+
+    options {
+        timeout(time: 240, unit: 'MINUTES')
+        timestamps()
+        buildDiscarder(logRotator(daysToKeepStr: '30', numToKeepStr: '30'))
+        ansiColor('xterm')
+        disableConcurrentBuilds()   
+    }
+
+    parameters {
+        string(name: 'CORTX_RE_REPO', defaultValue: 'https://github.com/Seagate/cortx-re/', description: 'Repository for Cluster Setup scripts', trim: true)
+        string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch or GitHash for Cluster Setup scripts', trim: true)
+        text(defaultValue: '''hostname=<hostname>,user=<user>,pass=<password>''', description: 'VM details to be used. First node will be used as Primary node. Please provide a minimum of 3 nodes', name: 'hosts')
+        text(defaultValue: '''/dev/sdX''', description: 'Disks for OSD on VMs. This assumes the disk name and no. of disk is same on all nodes', name: 'OSD_Disks')
+    }    
+
+    stages {
+        stage('Checkout Script') {
+            steps { 
+                cleanWs()            
+                script {
+                    checkout([$class: 'GitSCM', branches: [[name: "${CORTX_RE_BRANCH}"]], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: "${CORTX_RE_REPO}"]]])                
+                }
+            }
+        }
+
+        stage ('Install Ceph Prerequisites') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                sh label: 'Install Ceph Prerequisites', script: '''
+                    pushd solutions/kubernetes/
+                        echo $hosts | tr ' ' '\n' > hosts
+                        cat hosts
+                        bash ceph-deploy.sh --install-prereq
```

Comment:
What is the difference between `--install-prereq` and `--deploy-prereq`

Reply:
`--install-prereq` Before installing ceph packages some dependencies needs to be installed.
`--deploy-prereq` Before deploying ceph, nodes b/w themselves require passwordless ssh to copy ceph configuration and keys.

Reply:
Can you please add these details to the `usage` function of script. For other options also add some more description.

Reply:
Done